src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -1157,5 +1184,30 @@ private string BuildRedirectUriIfRelative(string uri)
 
             return BuildRedirectUri(uri);
         }
+
+        private OpenIdConnectProtocolException CreateOpenIdConnectProtocolException(OpenIdConnectMessage message, HttpResponseMessage response)
+        {
+            var description = message.ErrorDescription ?? "error_description is null";
+            var errorUri = message.ErrorUri ?? "error_uri is null";
+
+            if (response != null)
+            {
+                Logger.AuthorizationResponseErrorWithStatusCode(message.Error, description, errorUri, (int)response.StatusCode);
```

Comment:
I guess @Tratcher means the event is badly named: the response you receive at this point is a token response, not an authorization response.